src/main/kotlin/com/openlattice/indexing/BackgroundExpiredDataDeletionService.kt

```diff
@@ -0,0 +1,250 @@
+package com.openlattice.indexing
+
+import com.google.common.base.Stopwatch
+import com.hazelcast.core.HazelcastInstance
+import com.hazelcast.core.IMap
+import com.hazelcast.query.Predicate
+import com.hazelcast.query.Predicates
+import com.hazelcast.query.QueryConstants
+import com.openlattice.conductor.rpc.ConductorElasticsearchApi
+import com.openlattice.data.DataExpiration
+import com.openlattice.data.storage.IndexingMetadataManager
+import com.openlattice.data.storage.PostgresEntityDataQueryService
+import com.openlattice.edm.EntitySet
+import com.openlattice.edm.set.ExpirationType
+import com.openlattice.edm.type.EntityType
+import com.openlattice.edm.type.PropertyType
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.indexing.configuration.IndexerConfiguration
+import com.openlattice.postgres.DataTables.LAST_WRITE
+import com.openlattice.postgres.PostgresColumn.*
+import com.openlattice.postgres.PostgresTable.DATA
+import com.openlattice.postgres.PostgresTable.IDS
+import com.openlattice.postgres.ResultSetAdapters
+import com.openlattice.postgres.streams.PostgresIterable
+import com.openlattice.postgres.streams.StatementHolder
+import com.zaxxer.hikari.HikariDataSource
+import org.apache.olingo.commons.api.edm.EdmPrimitiveTypeKind
+import org.slf4j.LoggerFactory
+import org.springframework.scheduling.annotation.Scheduled
+import java.sql.Types
+import java.sql.PreparedStatement
+import java.sql.ResultSet
+import java.time.Instant
+import java.time.LocalDate
+import java.time.OffsetDateTime
+import java.time.ZoneId
+import java.util.*
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.locks.ReentrantLock
+import java.util.function.Function
+import java.util.function.Supplier
+
+/**
+ *
+ */
+
+const val MAX_DURATION_MILLIS = 60000L
+const val DATA_DELETION_RATE = 30000L
+
+class BackgroundExpiredDataDeletionService(
+        hazelcastInstance: HazelcastInstance,
+        private val indexerConfiguration: IndexerConfiguration,
+        private val hds: HikariDataSource,
+        private val elasticsearchApi: ConductorElasticsearchApi
+) {
+    companion object {
+        private val logger = LoggerFactory.getLogger(BackgroundExpiredDataDeletionService::class.java)!!
+    }
+
+    private val propertyTypes: IMap<UUID, PropertyType> = hazelcastInstance.getMap(HazelcastMap.PROPERTY_TYPES.name)
+    private val entitySets: IMap<UUID, EntitySet> = hazelcastInstance.getMap(HazelcastMap.ENTITY_SETS.name)
+    private val expirationLocks: IMap<UUID, Long> = hazelcastInstance.getMap(HazelcastMap.EXPIRATION_LOCKS.name)
+
+    //necessary??????
+    init {
+        expirationLocks.addIndex(QueryConstants.THIS_ATTRIBUTE_NAME.value(), true)
```

Comment:
I feel like this should be somewhere else but unsure where 🤔

Reply:
I think this is left over from the code I copied from BackgroundIndexingService, will remove